pyqtgraph/graphicsItems/AxisItem.py

```diff
@@ -1159,6 +1165,11 @@ def generateDrawSpecs(self, p):
                 textFlags = alignFlags | QtCore.Qt.TextFlag.TextDontClip    
                 #p.setPen(self.pen())
                 #p.drawText(rect, textFlags, vstr)
+
+                br = self.boundingRect()
+                if not br.contains(rect):
```

Comment:
I just realized, that my note in the commit message 
> With 'hideOverlappingLabels==False' nothing changes.

isn't true. Originally, the labels get clipped, if they go over the bounding rect. After my change, they will be hidden, independent of `hideOverlappingLabels`.
So, to get really the same behaviour as originally with `hideOverlappingLabels==False`, one should change the following.

```suggestion
                if self.style['hideOverlappingLabels'] and not br.contains(rect):
```

But, as painting outside of the bounding rect is not allowed (see https://doc.qt.io/qt-5/qgraphicsitem.html#paint), this clipping should additionally be implemented or we simply keep the hiding enabled if it goes out ot the bounding rect.

Reply:
Given that painting outside boundingRect I'm good w/ how it currently is